### PR TITLE
Instant search

### PR DESCRIPTION
**Pull request by alexey-sevastynov**
added the “Select” component to the “InstantSearch” section component, an adaptation for the “InstantSearch” viewing tablet.

### CODE REVIEW

**Comment by alexey-sevastynov** [comment]
I can't understand the label

image

**Comment by GGO-web** [comment]
This is a "placeholder" prop for select, I just added it to the component, check it out

File: src/components/CustomPlayer/CustomPlayer.tsx [hunk 1]
```diff
@@ -67,6 +67,7 @@ export const CustomPlayer = () => {
           items={seasons}
           selectedKey={season}
           isLoading={isPlayerLoading || !seasons}
+          maxWidth={150}
```
Comment (GGO-web) [inline src/components/CustomPlayer/CustomPlayer.tsx:70]: why do we need maxWidth here?

File: src/components/Searchbar/InstantSearch.tsx [hunk 2]
```diff
@@ -21,37 +27,62 @@ export const InstantSearch = () => {
     setSearch(e.target.value);
 
     await axios
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:29]: You can try to add loading when using async requests

File: src/components/Searchbar/InstantSearch.tsx [hunk 3]
```diff
@@ -21,37 +27,62 @@ export const InstantSearch = () => {
...
+      })
+      .catch((error) => {
+        if (axios.isCancel(error) || error) {
+          console.log('Could not get');
+        }
+      });
   };
 
   return (
     <div className="searchModule">
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:42]: rename the whole component InstantSearch is bad name and name of the classes too

File: src/components/Searchbar/InstantSearch.tsx [hunk 4]
```diff
@@ -21,37 +27,62 @@ export const InstantSearch = () => {
...
-      </ul>
+      <div className="searchModule__input ">
+        <input
+          value={search}
+          onChange={(e) => onLsChange(e)}
+          type="text"
+          placeholder="Почніть вводити назву..."
+        />
+
+        <ul>
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:51]: add class for the list and for the list items

File: src/components/Searchbar/InstantSearch.tsx [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
+import { Select } from '@components/Select/Select';
+import { IJsonAlbum } from '../../types/album';
+import { ANIME_GENRES } from '../../constants';
 
 export const InstantSearch = () => {
   const [posts, setPosts] = React.useState<IJsonAlbum[]>([]);
 
   const [search, setSearch] = useState('');
 
+  const [genr, setGenr] = useState<React.Key>(1);
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:15]: rename to genre not "genr" :)

File: src/components/Searchbar/InstantSearch.tsx [hunk 6]
```diff
@@ -21,37 +27,62 @@ export const InstantSearch = () => {
...
+            </Item>
+          )}
+        </Select>
       </div>
 
-      <div className="container">
-        <button className="button-style" type="button">Пошук</button>
-      </div>      
+      <button
+        style={{ height: '54px' }}
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:80]: check if this height is required

File: src/scss/screens/home/components/_search.scss [hunk 7]
```diff
@@ -31,6 +31,13 @@
     padding-left: 60px;
   }
 
+  &__select,
```
Comment (GGO-web) [inline src/scss/screens/home/components/_search.scss:34]: never add static width to elements, check this again and use flex property for flex items

File: src/scss/screens/home/components/_search.scss [hunk 8]
```diff
@@ -58,3 +65,20 @@
   border-color: rgb(41, 41, 41);
   background-color: rgb(41, 41, 41);
 }
+
+@media (max-width: $tablet) {
+  .search-block {
+    height: 200px;
```
Comment (GGO-web) [inline src/scss/screens/home/components/_search.scss:71]: static height is not good too :)

**Review by GGO-web — CHANGES_REQUESTED** [review]

File: src/components/Searchbar/InstantSearch.tsx (hunk 3 above)
Comment (alexey-sevastynov) [reply src/components/Searchbar/InstantSearch.tsx:45]: SearchBar.tsx or Search.tsx ??

File: src/components/Searchbar/InstantSearch.tsx (hunk 6 above)
Comment (alexey-sevastynov) [reply src/components/Searchbar/InstantSearch.tsx:89]: if you remove it, then: 
image

Comment (alexey-sevastynov) [reply src/components/Searchbar/InstantSearch.tsx:89]: made position absolute for block UL 
image

File: src/components/Searchbar/InstantSearch.tsx [hunk 9]
```diff
@@ -19,39 +26,73 @@ export const InstantSearch = () => {
...
-          if (axios.isCancel(error) || error) {
-            console.log('Could not get');
-          }
-        });
+      .get(`https://jsonplaceholder.typicode.com/albums`)
+      .then((res) => {
+        setPosts(res.data);
+        setIsSearchLoading(false);
+      })
+      .catch((error) => {
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:37]: If there are problems with api we will have infinity loading, so you can add here set loading to false also

File: src/components/Searchbar/InstantSearch.tsx (hunk 3 above)
Comment (GGO-web) [reply src/components/Searchbar/InstantSearch.tsx:45]: Just search will be okay

File: src/components/Searchbar/InstantSearch.tsx [hunk 10]
```diff
@@ -19,39 +26,73 @@ export const InstantSearch = () => {
...
-      </ul>
+      <div className="searchModule__input ">
+        <input
+          value={search}
+          onChange={(e) => onLsChange(e)}
+          type="text"
+          placeholder="Почніть вводити назву..."
+        />
+        {search.length > 0 &&
+          (isSearchLoading ? (
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:54]: there is duplicate of the list container elements, add this element inside a single wrapper in this case
<ul>
{isSearchLoading ? (<li>Loading....</li>): searchRes.map(....)}
</ul>

File: src/components/Searchbar/InstantSearch.tsx [hunk 11]
```diff
@@ -19,39 +26,73 @@ export const InstantSearch = () => {
...
-      </div>      
+      <div className="searchModule__select">
+        <Select
+          label="Виберіть епізод"
+          items={ANIME_GENRES}
+          selectedKey={genre}
+          placeholder="Виберіть жанр"
+          bgNone
+          maxWidth={250}
+          isLoading={isGenreLoading || !ANIME_GENRES}
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:79]: Just isGenreLoading here

File: src/components/Searchbar/InstantSearch.tsx [hunk 12]
```diff
@@ -19,39 +26,73 @@ export const InstantSearch = () => {
...
+          }}
+        >
+          {(item) => (
+            <Item textValue={item.title} id={item.id}>
+              {item.title}
+            </Item>
+          )}
+        </Select>
+      </div>
+      <div className="searchModule__button">
```
Comment (GGO-web) [inline src/components/Searchbar/InstantSearch.tsx:91]: Please remove this wrapper element and move the class to the button instead

File: src/components/SliderIMDB/SliderIMDB.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  thumbLabels?: string[];
+}
+
+export function SliderIMDB<T extends number | number[]>({
+  label,
+  thumbLabels,
+  ...props
+}: SliderIMDBProps<T>) {
+  return (
+    <Slider {...props} minValue={0} maxValue={10} step={0.1}>
```
Comment (GGO-web) [inline src/components/SliderIMDB/SliderIMDB.tsx:23]: add default props for this attributes like:
({
label,
thumbLabels,
minValue = 0,
maxValue = 0,
step = 0.1
})

**Review by GGO-web — CHANGES_REQUESTED** [review]

**Review by GGO-web — APPROVED** [review]

**Review by GGO-web — APPROVED** [review]

**Pull request merged**